lib/elixir/lib/calendar/datetime.ex

```diff
@@ -1829,6 +1829,50 @@ defmodule DateTime do
     shift(datetime, Duration.new!(duration), time_zone_database)
   end
 
+  @doc """
+  Returns the current datetime in UTC subtracting the given `duration`.
+
+  ## Examples
+
+      iex> ndt = DateTime.utc_ago(day: 1)
+      iex> ndt == DateTime.add(DateTime.utc_now(), -1, :day)
+
+      iex> ndt = DateTime.utc_ago(month: 1)
+      iex> ndt == DateTime.shift(DateTime.utc_now(), month: -1)
+
+  """
+  @doc since: "1.17.0"
+  @spec utc_ago(Duration.t() | [Duration.unit_pair()]) :: t
+  def utc_ago(%Duration{} = duration) do
```

Comment:
there is no rush but curious if we should add a `precision \\ :microsecond` to mimic `DateTime.utc_now(:second)`.

it would be a bit awkward with keyword argument though:

```elixir
DateTime.utc_ago([hour: 42], :second)
```

so maybe it’d be:

```elixir
DateTime.utc_ago(hour: 42, precision: :second)
```

and so probably no Calendar.duration() type after all. 😅

Reply:
We could be adding both arity `/1` and `/2` versions

```
DateTime.utc_ago(%Duration{hour: 42}, :second)
```

or

```
DateTime.utc_ago(hour: 42, precision: :second)
```